tests/store/crud/test_update.py

```diff
@@ -366,3 +368,111 @@ def test_update_analysis_from_slurm_run_status(
+
+    # THEN jobs should be updated
+    assert len(analysis.jobs) == 2
+
+
+@pytest.mark.parametrize(
+    "case_id, status, progress",
+    [
+        (CaseId.RUNNING, TrailblazerStatus.RUNNING, 0.15),
+        (CaseId.PENDING, TrailblazerStatus.PENDING, 0),
+        (CaseId.COMPLETED, TrailblazerStatus.QC, 1),
+    ],
+)
+def test_update_tower_run_status(
+    analysis_store: MockStore,
+    case_id: str,
+    status: str,
+    progress: int,
+    mocker,
+    tower_case_config: Dict[str, dict],
+):
+    """Assess that an analysis status is successfully updated when using NF Tower."""
+
+    # GIVEN Tower API response for an analysis
+    raw_case: dict = tower_case_config.get(case_id)
+    tower_api = MockTowerAPI(workflow_id=raw_case.get("tower_id"))
+    tower_api.mock_query(response_file=raw_case.get("workflow_response_file"))
+    tower_api.mock_tasks_query(response_file=raw_case.get("tasks_response_file"))
+    mocker.patch("trailblazer.store.crud.update.get_tower_api", return_value=tower_api)
+
+    # GIVEN an analysis with pending status
+    analysis: Optional[Analysis] = analysis_store.get_latest_analysis_for_case(case_id=case_id)
+    assert analysis.status == TrailblazerStatus.PENDING
+
+    # WHEN the database is updated once
+    analysis_store.update_tower_run_status(analysis_id=analysis.id)
+
+    # THEN analysis status is updated
+    assert analysis.status == status
+    assert analysis.progress == progress
+
+    # WHEN the database is updated a second time
+    analysis_store.update_tower_run_status(analysis_id=analysis.id)
+
+    # THEN the status is unchanged, and no database errors were raised
+    assert analysis.status == status
+    assert analysis.progress == progress
+
+
+@pytest.mark.parametrize(
+    "case_id, status, progress",
+    [
+        (CaseId.RUNNING, TrailblazerStatus.RUNNING, 0.15),
+        (CaseId.PENDING, TrailblazerStatus.PENDING, 0),
+        (CaseId.COMPLETED, TrailblazerStatus.QC, 1),
+    ],
+)
+def test_update_run_status_using_tower(
+    analysis_store: MockStore,
+    case_id: str,
+    status: str,
+    progress: int,
+    mocker,
+    tower_case_config: Dict[str, dict],
+):
+    """Assess that an analysis status is successfully updated when using NF Tower."""
+
+    # GIVEN Tower API response for an analysis
+    raw_case: dict = tower_case_config.get(case_id)
+    tower_api = MockTowerAPI(workflow_id=raw_case.get("tower_id"))
+    tower_api.mock_query(response_file=raw_case.get("workflow_response_file"))
+    tower_api.mock_tasks_query(response_file=raw_case.get("tasks_response_file"))
+    mocker.patch("trailblazer.store.crud.update.get_tower_api", return_value=tower_api)
+
+    # GIVEN an analysis with pending status
+    analysis: Optional[Analysis] = analysis_store.get_latest_analysis_for_case(case_id=case_id)
+    assert analysis.status == TrailblazerStatus.PENDING
+
+    # WHEN the database is updated once
+    analysis_store.update_run_status(analysis_id=analysis.id)
```

Comment:
The function `update_run_status` has not yet been moved from `store/api` to `crud/update`. Maybe this test should stay in `test_store_api.py`, or the function `update_run_status` could be moved to `crud/update`.

Reply:
I just saw that the function was moved in #291 so disregard this comment. But still, I think this test function could be refactored as I suggested above

Reply:
Fixed